core/src/Kokkos_HPX.hpp

```diff
@@ -202,33 +196,42 @@ class thread_buffer {
 
 namespace Experimental {
 class HPX {
+ public:
+  static constexpr uint32_t impl_default_instance_id() { return 1; }
+
  private:
   static bool m_hpx_initialized;
-  static std::atomic<uint32_t> m_next_instance_id;
-  uint32_t m_instance_id = 0;
+  uint32_t m_instance_id = impl_default_instance_id();
 
 #if defined(KOKKOS_ENABLE_HPX_ASYNC_DISPATCH)
+  static std::atomic<uint32_t> m_next_instance_id;
+
  public:
-  enum class instance_mode { global, independent };
-  instance_mode m_mode;
+  enum class instance_mode { default_, independent };
```

Comment:
Because `default` is a keyword. :upside_down_face:. I changed the name because I had the impression that "default" better fits the language that you use for the (...) default instances, rather than global. I'm not set on `default_` though and I might have to come up with a better alternative at some point.

Reply:
Oh right